lib/apiservers/engine/backends/container.go

```diff
@@ -808,13 +811,16 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 			}
 			return InternalServerError(err.Error())
 		}
-		// force stop if container state is error to make sure container is deletable later
-		if state.Status == ContainerError {
+
+		switch state.Status {
+		case ContainerError:
+			// force stop if container state is error to make sure container is deletable later
 			c.containerProxy.Stop(vc, name, &secs, true)
-		}
-		// if we are starting let the user know they must use the force
-		if state.Status == "Starting" {
+		case "Starting":
+			// if we are starting let the user know they must use the force
 			return derr.NewRequestConflictError(fmt.Errorf("The container is starting.  To remove use -f"))
+		case ContainerRunning:
+			running = true
```

Comment:
In this case we're not using force remove, so we shouldn't be deleting the container at all.
Why not return here with an error saying the container is running.

Reply:
My first version did return from here, but I thought back to our policy of attempting to perform portlayer actions and letting it fail.  We could simply return the error here.